drivers/gpio/gpio_silabs.c

```diff
@@ -315,6 +366,25 @@ static int gpio_silabs_port_manage_callback(const struct device *dev, struct gpi
 	return gpio_manage_callback(&data->callbacks, cb, set);
 }
 
+static void gpio_silabs_em4wu_interrupt_to_port_pin(struct gpio_silabs_common_data *data,
+						    int int_no, int *port, int *pin)
+{
+	ARRAY_FOR_EACH(data->ports, p) {
+		const struct device *dev = data->ports[p];
+		const struct gpio_silabs_port_config *config = dev->config;
+
+		for (int i = 0; i < config->em4wu_pin_count; i++) {
+			const struct gpio_silabs_em4wu_mapping *em4wu = &config->em4wu_pins[i];
+
+			if (em4wu->int_no == (int_no - NUM_IRQ_LINES)) {
```

Comment:
Same here. Did you consider having helper macros for the translation (where the macro name would help indicate what you're doing)?

Reply:
Introduced a macro and improved the variable names to make it more self documenting.